.github/workflows/client-api-tests-and-coverage.yml

```diff
@@ -131,7 +131,7 @@ jobs:
   client-unit-tests:
     env:
       COVERAGE_FILE: .coverage.client-unit-tests
-    runs-on: ubuntu-latest
+    runs-on: ubuntu-22.04
```

Comment:
where are your runners hosted? just using github ones? why wouldn't you use `24.04`?

Reply:
We are using the github ones. It was previously pinned to `ubuntu-latest` which I am assuming was `22.04` until this weekend. 

The reason for this chnage is that `python3.7` is not a valid `24.04` target.

Reply:
why do we care? python 3.7 is EOL

Reply:
Im good with removing it, the valor-client code is not relevant to this pr so i just forced that version.